### ISSUE TITLE
gitit won't install on NixOS 20.09

### ISSUE THREAD

**Issue by mhwombat** [NixOS/nixpkgs#107209]
**Describe the bug**
`gitit` won't install on NixOS 20.09

**To Reproduce**
1. Add `gitit` to `environment.systemPackages`.
2. `nixos-rebuild switch --upgrade`
3. Rebuild fails with error messages
4. Remove `gitit` from `environment.systemPackages`.
5. `nixos-rebuild switch --upgrade`
6. Rebuild works

**Expected behavior**
I expect the rebuild to complete without errors.

**Screenshots**

```
# nixos-rebuild switch --upgrade
unpacking channels...
created 1 symlinks in user environment
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/9xmri1cxv3ha2ki57x1m0p86jffmr13f-gitit-0.13.0.0.drv
  /nix/store/a8d7c13dkxp26j7j0d08ni6fwf3f3lqz-system-path.drv
  /nix/store/d5lgrm3w2awx7zvds5caakysvb0jkkd1-dbus-1.drv
  /nix/store/0d9fqj1jil54dpb42qmc0xv5vwsizs6g-unit-dbus.service.drv
  /nix/store/6w4nlrb6ihzzdsmba3l3gadqimrsj7p1-unit-systemd-fsck-.service.drv
  /nix/store/wywg52jifw2jg3x83pirdxc25ab8cw3n-unit-polkit.service.drv
  /nix/store/4z2dxi30pdpbzkkppibxigl4mbvnhcpx-system-units.drv
  /nix/store/cddy9975dy5ai62pn51v62x2pwz6z1g8-unit-dbus.service.drv
  /nix/store/ns2kfgki22ypnrk1p7m0chg3xv1aqig8-user-units.drv
  /nix/store/0qzy9wg90rphniwf6p78ycr7jwm0hmq0-etc.drv
  /nix/store/ljfwdz7zpv2bnxc7lrl33lfac0xygixq-nixos-system-nixos-20.09.2290.647cc06986c.drv
building '/nix/store/9xmri1cxv3ha2ki57x1m0p86jffmr13f-gitit-0.13.0.0.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/8wp3hwg7nnxq47n4vpj83wxl3pmmpk19-ghc-8.8.4.
unpacking sources
unpacking source archive /nix/store/pxqzbfr2pvw0ccmvb1r76fawr79a0mdl-gitit-0.13.0.0.tar.gz
source root is gitit-0.13.0.0
setting SOURCE_DATE_EPOCH to timestamp 1591212277 of file gitit-0.13.0.0/gitit.cabal
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/build/setup-package.conf.d -j16 +RTS -A64M -RTS -threaded -rtsopts
[1 of 1] Compiling Main             ( Setup.lhs, /build/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/68vd91hnzldz003h28w17i67h4gaalyb-gitit-0.13.0.0 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --datadir=/nix/store/4f8x3isyzay7c43l66wd8fnbnc6b6pqq-gitit-0.13.0.0-data/share/ghc-8.8.4 --with-gcc=gcc --package-db=/build/package.conf.d --ghc-options=-j16 +RTS -A64M -RTS --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections -f-plugins --extra-lib-dirs=/nix/store/qxd21xy8mz37x9a2cwj7zaw8813a1c8q-ncurses-6.2/lib --extra-lib-dirs=/nix/store/mqr1hbh7jmgpmdfd7bb0yr54brb1b9xy-libffi-3.3/lib --extra-lib-dirs=/nix/store/ks724y9k5skmsr5y4gii28nfqrb1r5bj-gmp-6.2.0/lib --extra-lib-dirs=/nix/store/8wp3hwg7nnxq47n4vpj83wxl3pmmpk19-ghc-8.8.4/lib
Using Parsec parser
Configuring gitit-0.13.0.0...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1022:20 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:475:12 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.0.1.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.0.1.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.0.1.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.0.1.0:Distribution.Simple
  defaultMain, called at Setup.lhs:3:10 in main:Main
Setup: Encountered missing or private dependencies:
hoauth2 >=1.3.0 && <1.12, pandoc ==2.9.*, pandoc-types ==1.20.*

builder for '/nix/store/9xmri1cxv3ha2ki57x1m0p86jffmr13f-gitit-0.13.0.0.drv' failed with exit code 1
cannot build derivation '/nix/store/a8d7c13dkxp26j7j0d08ni6fwf3f3lqz-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ljfwdz7zpv2bnxc7lrl33lfac0xygixq-nixos-system-nixos-20.09.2290.647cc06986c.drv': 1 dependencies couldn't be built
error: build of '/nix/store/ljfwdz7zpv2bnxc7lrl33lfac0xygixq-nixos-system-nixos-20.09.2290.647cc06986c.drv' failed

```

**Additional context**

**Notify maintainers**

@Profpatsch 

**Metadata**

```
# nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/ajidkqg53y0zvh0m5cjgqiw3pf9cvlf0-bash-interactive-4.4-p23-dev
copying path '/nix/store/ajidkqg53y0zvh0m5cjgqiw3pf9cvlf0-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.80, NixOS, 20.09.2290.647cc06986c (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-20.09.2290.647cc06986c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: gitit
# a list of nixos modules affected by the problem
module: 
```

**Comment by sternenseemann** [comment]
Should be fixed in a3a3dda3bacf61e8a39258a0ed9c924eeca8e293, might take some time till it hits the `nixos-20.09` channel though.

**Comment by sternenseemann** [comment]
[Build on Hydra succeeded](https://hydra.nixos.org/build/133381100#tabs-summary) and fix in nixos-20.09. Should work after a `nix-channel --update`. Feel free to reopen if the problem persists.

**Comment by siraben** [comment]
@mhwombat BTW, it's possible to test building of individual packages without changing the system config, try

```ShellSession
$ nix build -f '<nixpkgs>' gitit
```
